web/packages/teleterm/src/ui/AppInitializer.tsx

```diff
@@ -14,29 +14,37 @@
  * limitations under the License.
  */
 
-import React, { useCallback, useEffect, useState } from 'react';
+import React, { useCallback, useEffect, useState, useRef } from 'react';
 import styled from 'styled-components';
 import { Indicator } from 'design';
 
+import Logger from 'teleterm/logger';
+
 import { useAppContext } from './appContextProvider';
 import { initUi } from './initUi';
 import ModalsHost from './ModalsHost';
 import { LayoutManager } from './LayoutManager';
 
 export const AppInitializer = () => {
-  const ctx = useAppContext();
+  const loggerRef = useRef<Logger>(null);
+  if (loggerRef.current === null) {
+    loggerRef.current = new Logger('AppInitializer');
+  }
+
+  const appContext = useAppContext();
   const [isUiReady, setIsUiReady] = useState(false);
 
   const initializeApp = useCallback(async () => {
     try {
-      await ctx.init();
-      await initUi(ctx);
+      await appContext.init();
+      await initUi(appContext);
       setIsUiReady(true);
     } catch (error) {
+      loggerRef.current.error(error?.message);
       setIsUiReady(true);
-      ctx.notificationsService.notifyError(error?.message);
+      appContext?.notificationsService.notifyError(error?.message);
```

Comment:
This was changed because technically `appContext.init` can throw before it sets up `notificationsService`.